lib/dal.js

```diff
@@ -0,0 +1,76 @@
+var Q = require('q');
+var fetchJSON = require('./fetchJSON.js');
+
+/** Attempts to access configuration file data for the shared extension
+ * @param {string} confFile - the path and filename to the configuration file
+ * @param {string} name - the name of the sharedExtension to access
+ * @return {Q} - Q promise object
+ */
+
+var getSharedExtension = function(confFile, name) {
+  var defer = Q.defer();
+  var sharedExtension = {};
```

Comment:
From a stylistic perspective, it is often nice to have a line break between your variable declarations and the code that acts on them.

```javascript
var defer = Q.defer();
var sharedExtension = {};

sharedExtension.found = false;
...
```